gio/src/dbus_node_info.rs

```diff
@@ -0,0 +1,38 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::ffi::CStr;
+
+use glib::translate::ToGlibPtr;
+
+use crate::{DBusInterfaceInfo, DBusNodeInfo};
+
+impl DBusNodeInfo {
+    pub fn path(self) -> &'static str {
+        let c_obj = self.to_glib_none().0;
+        let path = unsafe { (*c_obj).path };
+        if path.is_null() {
+            return "";
+        }
+        let c_str = unsafe { CStr::from_ptr(path) };
+        c_str.to_str().unwrap()
+    }
+
+    pub fn interfaces(self) -> Vec<DBusInterfaceInfo> {
```

Comment:
```suggestion
    pub fn interfaces<'a>(&'a self) -> &'a [DBusInterfaceInfo] {
```

via https://gtk-rs.org/gtk-rs-core/git/docs/glib/collections/struct.PtrSlice.html#method.from_glib_borrow_num should also work. We should add a non-`_num` version of this.

Reply:
Done